faunadb-android/src/test/java/com/faunadb/client/AndroidClientSpec.java

```diff
@@ -51,4 +68,60 @@ public void shouldAuthenticateSession() throws Exception {
     assertThat(identified.to(BOOLEAN).get(), is(false));
   }
 
-}
\ No newline at end of file
+  @Before
+  public void setUpClient() throws Exception {
+    if (rootClient != null)
+      return;
+
+    rootClient = createFaunaClient(ROOT_TOKEN);
+
+    client = transform(
+      transformAsync(
+        setupDatabase(DB_REF, DB_NAME),
```

Comment:
I don't think if I understood, it's not possible to move (I think) things up to `DslSpec` because in order to setup database we need a `rootClient` and in order create the `client` we need a database